consensus/polybft/bridge/bridge_manager.go

```diff
@@ -107,6 +111,9 @@ type bridgeEventManager struct {
 	lock                         sync.RWMutex
 	pendingBridgeBatchesExternal []*PendingBridgeBatch
 	pendingBridgeBatchesInternal []*PendingBridgeBatch
+	unexecutedBatches            []*PendingBridgeBatch
```

Comment:
I really do not like this approach. I know we agreed that we will rollback the entire batch, but that means that we are over-complicating the client (validators) because of this. They need to keep track of executed batches, and transactions that have or not have been executed successfully on some chain (which is the job a relayer), rather than just listening to `BridgeMessageResult` events.

@Nemanja0x @oliverbundalo Please reconsider the approach of executing whatever we can in the `BridgeBatch` transaction, meaning it should never fail. In my point of view, to do this is pretty easy, and won't require a lot of changes:

1. Do not revert entire batch if some `BridgeMessage` fails.
2. Successful bridge messages will emit a `BridgeMessageResult` event with a status flag to true (to mark they have been executed successfully), and the ones that fail, will emit a `BridgeMessageResult` event with a status flag set to false (something similar we had in the original bridge).
3. Validators will keep track of the failed messages.
4. Like building regular batches, on each block, they will check for failed bridge message events they received from event tracker/provider, and build rollback batches from them as well (we already have map of processed rollback events, so we now which were executed and which were not).
5. This falls under the same mechanism as the regular batches, and I do not see big issues, or big changes required on the validators, and to me, the logic is much simpler.

Reply:
There is a possibility that initial implementation (not reverting entire batch) will be adopted again as a final solution. However, even if we do so there is a still theoretical chance that the batch is not executed at all at the destination chain and in that case we will receive no events whatsoever. So IMO we still need some kind of logic to support that case, either this way or some another.

Reply:
Nope. We do not need any logic around that im the validators. That will be a case for the retry mechanism in the relayer.